aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -403,6 +403,119 @@ class TypedOperatorHandle<Return (Args...)> final : public OperatorHandle {
 
 namespace detail {
 template<class... Args> inline void unused_arg_(const Args&...) {}
+
+// ReturnValue is intended to capture return values from Dispatcher unboxed
+// kernel calls.
+// A record function may requests to get outputs from the kernel calls. For
+// boxed kernels, it's straightforward, the returned values are in the stack
+// object. The stack can be passed to record functions. For unboxed kernels, we
+// need to handle different kinds of return values, cache them temporarily, then
+// release the values for the actual function call return.
+template <typename Return>
+struct ReturnValue {
```

Comment:
@smessmer @swolchok Sebastian, Scott - any ideas about adding this template?

Can we just reuse the existing code for boxing in boxing/impl/boxing.h ?

Reply:
Thanks, trying a few ideas, will follow up.